### PR TITLE
Added error codes to endpoints and fixed HTTP resp codes.

### PR DESCRIPTION
**Pull request by KunalGhosh02**
Fixed issue #12 and #13 
Did not make any changes to verification route because front-end uses it's legacy forms.

### CODE REVIEW

File: src/routes/contact.js [hunk 1]
```diff
@@ -57,14 +57,14 @@ router.delete('/', auth, async (req, res) => {
 router.get('/', auth, async (req, res) => {
 	try {
 		const user = await User.findOne({ uid: req.user.uid }).select('-password');
-		if (user.role != 'admin') res.status(401).json({ msg: 'Not authorized' });
+		if (user.role != 'admin') res.status(401).json({ code: 'QUERY_FETCH_UNAUTHORISED', msg: 'Not authorized' });
 		else {
 			const dContact = await Contact.find({});
 			if (dContact) res.json(dContact);
-			else res.json({ msg: `Error in fetching` });
+			else res.status(417).json({ code: 'QUERY_FETCH_ERROR', msg: `Error in fetching` });
```
Comment (lazycipher) [inline src/routes/contact.js]: Why `417`?

Comment (KunalGhosh02) [reply src/routes/contact.js]: Missed that it does not take a headers, fixed.

File: src/routes/events.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ router.get('/:eventId', async (req, res) => {
 	try {
 		const event = await Events.findOne({ eid: req.params.eventId });
 		if (!event) {
-			res.status(404).send('Resource Not Found');
+			res.status(404).json({ code: 'EVENTS_EVENTID_NOT_FOUND', msg: 'Resource Not Found' });
```
Comment (lazycipher) [inline src/routes/events.js]: `EVENT_NOT_FOUND` won't look better?

File: src/routes/events.js [hunk 3]
```diff
@@ -59,13 +59,13 @@ router.post('/add', auth, async (req, res) => {
 		const user = await User.findOne({ uid: req.user.uid }).select('-password');
 		console.log('user:', user);
 		console.log('uid:', req.user.uid);
-		if (user.role != 'admin') res.status(401).json({ msg: 'Not authorized' });
+		if (user.role != 'admin') res.status(401).json({ code: 'EVENTS_ADD_UNAUTHORISED', msg: 'Not authorized' });
 		else {
 			let eventCreation = await Events.create(event);
 			if (eventCreation) {
-				return res.status(201).json({ msg: 'Event Added Successfully' });
+				return res.status(201).json({ code: 'EVENTS_ADD_DONE', msg: 'Event Added Successfully' });
```
Comment (lazycipher) [inline src/routes/events.js]: `EVENT_ADD_SUCCESS` will look better!

File: src/routes/events.js [hunk 4]
```diff
@@ -59,13 +59,13 @@ router.post('/add', auth, async (req, res) => {
...
 			} else {
-				return res.status(400).json({ msg: 'Failed: Add Event Operation' });
+				return res.status(400).json({ code: 'EVENTS_ADD_FAILED', msg: 'Failed: Add Event Operation' });
```
Comment (lazycipher) [inline src/routes/events.js]: It'll be `EVEN`T not `EVENTS` as we're doing everything on a single event only!

File: src/routes/events.js [hunk 5]
```diff
@@ -84,7 +84,7 @@ router.put('/register/:eventId/', auth, async (req, res) => {
 			'users'
 		);
 		if (registeredUsers.users.includes(userId)) {
-			return res.status(400).json({ msg: 'User Already Registered' });
+			return res.status(400).json({ code: 'EVENTS_REGISTER_REDUNDANT', msg: 'User Already Registered' });
```
Comment (lazycipher) [inline src/routes/events.js]: Make this code something else!

File: src/routes/register.js [hunk 6]
```diff
@@ -17,12 +17,12 @@ const ADMIN_EMAIL = process.env.ADMIN_EMAIL;
...
 router.post('/', async (req, res) => {
 	const { name, email, password } = sanitize(req.body);
-	console.log(name,email,password)
+	console.log(name, email, password)
 	try {
 		let user = await User.findOne({ email });
 
 		if (user) {
-			return res.status(400).json({ msg: 'User already exists' });
+			return res.status(400).json({ code: 'REGISTER_CREATE_REDUNDANT', msg: 'User already exists' });
```
Comment (lazycipher) [inline src/routes/register.js]: Change this  too!

File: src/routes/register.js [hunk 7]
```diff
@@ -39,22 +39,22 @@ router.post('/', async (req, res) => {
...
+		var mailOptions = {
+			from: `${ADMIN_EMAIL}`,
+			to: user.email,
+			subject: 'TechnoJam Account Verification',
+			text: 'Team TechnoJam Welcomes You,\n\n' + `Your Verification Code: ${token.token}\n\nPlease verify your account by clicking the link and submitting the verification code: \nhttp:\/\/` + req.headers.host + '\/api\/verification\/' + token.token + '.\n\n' + 'Thank You,\nTeam TechnoJam'
 		};
 		transporter.sendMail(mailOptions, (err) => {
 			if (err) { return res.status(500).send({ msg: err.message }); }
-			res.status(201).json({msg: 'Account Created! Please check your mail to confirm account.'});
+			res.status(201).json({ code: 'REGISTER_CREATE_DONE', msg: 'Account Created! Please check your mail to confirm account.' });
```
Comment (lazycipher) [inline src/routes/register.js]: Looks odd! Make this something else like `USER_CREATED` or `SIGNUP_SUCCESS` or something like that.

**Review by lazycipher — COMMENTED** [review]
Just change things suggested and we'll be good to go!
Don't forget to change all single event request occurrences from `EVENTS` to `EVENT` and similarly, if there's anything that needs to be changed.

File: src/routes/events.js (hunk 2 above)
Comment (KunalGhosh02) [reply src/routes/events.js]: Done.
Ref: 6a2ddc7ac52071efdfeddcaf94fc721f8e70d1ff

File: src/routes/events.js (hunk 3 above)
Comment (KunalGhosh02) [reply src/routes/events.js]: Done.
Ref: 6a2ddc7ac52071efdfeddcaf94fc721f8e70d1ff

File: src/routes/events.js (hunk 4 above)
Comment (KunalGhosh02) [reply src/routes/events.js]: Done.
Ref: 6a2ddc7ac52071efdfeddcaf94fc721f8e70d1ff

File: src/routes/events.js (hunk 5 above)
Comment (KunalGhosh02) [reply src/routes/events.js]: Done.
Ref: 6a2ddc7ac52071efdfeddcaf94fc721f8e70d1ff

**Comment by KunalGhosh02** [comment]
I think everything is resolved.
Reference:
6a2ddc7ac52071efdfeddcaf94fc721f8e70d1ff
 3d08e9dcf8831ad1072222822448791ed774e54d

**Comment by lazycipher** [comment]
Replace all `______DONE` with `_____SUCCESS` and we're good to go!
btw have you worked with API Documentation using https://swagger.io/ or anything else?

**Comment by KunalGhosh02** [comment]
I haven't worked with API Documentation tools but I will look into it.

**Review by lazycipher — APPROVED** [review]
lgtm

**Pull request merged**